internal/common/conversion/type_conversion_test.go

```diff
@@ -67,3 +68,31 @@ func TestMongoDBRegionToAWSRegion(t *testing.T) {
 		}
 	}
 }
+
+func TestSlice_SameBehavior(t *testing.T) {
+	tests := []struct {
+		ptr   *[]string
+		slice []string
+	}{
+		{nil, nil},
+		{&[]string{"hello", "there"}, []string{"hello", "there"}},
+	}
+	for _, test := range tests {
+		assert.Equal(t, test.slice, conversion.SlicePtrToSlice(test.ptr))
+		assert.Equal(t, test.ptr, conversion.NonEmptySliceToSlicePtr(test.slice))
+	}
+}
+
+func TestSlice_DifferentBehavior(t *testing.T) {
```

Comment:
in slice tests I'm focusing in the different behavior (because the "nonempty" part) instead of testing each method separately, so it's more clear the different behavior, testing each method in its test method it's more challenging to see the differences

Reply:
@gssbzn  this is what I was trying here instead of testing each method in its own test function. I did that at first but then I thought this would be better

Reply:
at the end there is only one method, there is a function test for it as you say